src/regex_manager.cpp

```diff
@@ -153,7 +165,8 @@ FilterResponse RegexManager::on_http_request(const TransactionParts& transaction
 
   RatedRegex* result = try_match(
 						ats_record_parts[TransactionMuncher::IP],
-						ats_record);
+						ats_record,
+						ats_record_parts[TransactionMuncher::HOST]);
```

Comment:
Should I instead just pass in IP, METHOD, URL, HOST and construct this `ats_record` inside `try_match()`? I'm never sure in C++ how much I'm supposed to care about wasting unneeded memory copies and constructor calls and whatever else :)

Reply:
Similarly, should these parameters be `const std::string&`?

Reply:
1) When you do `s = s1 + s2 + s3`, you end up with a lot of re-allocations and copies. 
2) The better way probably is to allocate once, and then copy once:
```
std::string ats_record;

ats_record.reserve(ats_record_parts[TransactionMuncher::METHOD].size() + 
  ats_record_parts[TransactionMuncher::URL].size() + 
  ats_record_parts[TransactionMuncher::HOST].size() 
  ats_record_parts[TransactionMuncher::UA].size() +
  + 1 + 3);
ats_record += ats_record_parts[TransactionMuncher::METHOD];
ats_record += '" ";
ats_record += ats_record_parts[TransactionMuncher::URL];
ats_record += " ";
ats_record += ats_record_parts[TransactionMuncher::HOST];
ats_record += " "
ats_record += ats_record_parts[TransactionMuncher::UA];
```
3) In general it's always better to use `const std::string&`. Otherwise it's an extra string copy on the stack for nothing.
4) I don't see a reason to modify `try_match()`. You need to construct a string one way or another. Unless you have other reasons.